File: ITP1/5_A.js
function Main(input){
  input = input.split('\n')
	let i = 0
	let flag = true;
	while(flag){
		const [h, w] = input[i].split(' ').map(Number)
		if(h === 0 && w === 0){
			flag = false;
			break;
		}
		let rec = ''
		for(let j = 0; j < h; j++){
			for(let k = 0; k < w; k++){
				rec += '#'
			}
			rec += '\n'
		}
		console.log(rec);
		i++
	}
}



Main(require("fs").readFileSync("/dev/stdin", "utf8"));

File: ALDS1/5_A.py
n = int(input())
A = list(map(int, input().split()))
q = int(input())
m = list(map(int, input().split()))

S = [0] * n

for i in range(n):
    S[i] = 0

s = set()


def rec(i):
    if i == n:
        t = 0
        for i in range(n):
            if S[i]:
                t += A[i]
        s.add(t)
        return
    rec(i + 1)
    S[i] = 1
    rec(i + 1)
    S[i] = 0


rec(0)

for i in range(q):
    ans = "yes" if m[i] in s else "no"
    print(ans)
